pkg/channel/distributed/dispatcher/dispatcher/handler.go

```diff
@@ -60,33 +52,49 @@ func (h *Handler) Cleanup(_ sarama.ConsumerGroupSession) error {
 	return nil // Nothing To Do As Of Yet
 }
 
-// ConsumerGroupHandler Lifecycle Method (Main processing loop, must finish when Messages() channel closes.)
+// ConsumerGroupHandler Lifecycle Method (Main processing loop, must finish when claim.Messages() channel closes.)
 func (h *Handler) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
 
-	// Extract The Relevant Knative Subscription Event URLs
+	// Extract The Destination URL From The Subscriber
 	var destinationURL *url.URL
 	if !h.Subscriber.SubscriberURI.IsEmpty() {
 		destinationURL = h.Subscriber.SubscriberURI.URL()
 	}
 
+	// Extract The Reply URL From The Subscriber
 	var replyURL *url.URL
 	if !h.Subscriber.ReplyURI.IsEmpty() {
 		replyURL = h.Subscriber.ReplyURI.URL()
 	}
 
+	// Validate The Subscriber's Delivery (Optional)
 	var deadLetterURL *url.URL
-	if h.Subscriber.Delivery != nil &&
-		h.Subscriber.Delivery.DeadLetterSink != nil &&
-		h.Subscriber.Delivery.DeadLetterSink.URI != nil &&
-		!h.Subscriber.Delivery.DeadLetterSink.URI.IsEmpty() {
-		deadLetterURL = h.Subscriber.Delivery.DeadLetterSink.URI.URL()
+	retryConfig := kncloudevents.NoRetries()
+	if h.Subscriber.Delivery != nil {
+
+		// Extract The DeadLetterSink From The Subscriber.Delivery
+		if h.Subscriber.Delivery.DeadLetterSink != nil &&
+			h.Subscriber.Delivery.DeadLetterSink.URI != nil &&
+			!h.Subscriber.Delivery.DeadLetterSink.URI.IsEmpty() {
+			deadLetterURL = h.Subscriber.Delivery.DeadLetterSink.URI.URL()
+		}
+
+		// Extract The RetryConfig From The Subscriber.Delivery (Defaults To NoRetries)
+		var err error
+		retryConfig, err = kncloudevents.RetryConfigFromDeliverySpec(*h.Subscriber.Delivery)
+		if err != nil {
+			h.Logger.Error("Failed To Parse RetryConfig From DeliverySpec - No Retries Will Occur", zap.Error(err))
+		} else {
+			h.Logger.Info("Successfully Parsed RetryConfig From DeliverySpec", zap.Any("RetryMax", retryConfig.RetryMax))
```

Comment:
Is it better to use zap.Int when the field type is known?

Reply:
yeah, definitely - good catch - will fix it.